packages/hooks/src/useDocumentEventListener.ts

```diff
@@ -0,0 +1,29 @@
+import { useEffect } from 'react';
+
+/**
+ * Hook to subscribe to an event listener.
+ * @param type Represents the event type to listen for.
+ * @param eventCallback Event listener callback function.
+ * @param enabled Determines whether or not the hook should run, defaults to true.
+ * @param options Optional parameter to specify options passed to the eventListener.
+ * @param dependencies Optional array to be passed to useEffect hook, such that the hook will only run if the array's values have changed.
+ */
+export default function useDocumentEventListener(
+  type: string,
+  eventCallback: EventListener,
+  options?: object,
+  dependencies?: Array<any>,
```

Comment:
We should probably default this to an empty array. This hook should only fire once really unless dependencies are explicitly set.

Reply:
That's a departure from the defaults of `useEffect` which seems like a bad thing to splinter from.

Reply:
Hmm... perhaps. I feel that in the majority of cases, we only will want to have this fire on mount, but perhaps having that be an explicit decision is okay.

Reply:
I'm so torn on this. Looking at [this hook that Brooke linked to](https://usehooks.com/useEventListener/) its default behavior is probably what you want 99% of the time. In fact I don't know when we wouldn't want this behavior.

i.e. re-run the event listeners if the 'type' or `enabled` field changes. We can't really do it on options as that'd be a new object every time, and may be tricky to memoize. Might have to do more useRef trickery. 

Wonder if the type signature for this should be `useDocumentEventListener(type, eventCallback, options, { dependencies, enabled, element = document })`. Such that the API is the same as `addEventListener` but you can configure it more in the last arg.

Reply:
Perhaps `element` isn't the best name for that option, but I see your point. Hmm... we're deviating a bit from the argument pattern used for memoized hooks with including `dependencies`, but perhaps that breaks the expectation it's following a similar signature and allows us to have different defaults.

Reply:
The thing about including dependencies is that we could assume them, but where we're invoking this hook we're deviating from what would be the assumed dependencies so we'd be hurting ourselves and also limiting future flexibility in that way.

Reply:
I like the signature proposed by @hswolff, and I don't think element is that problematic because if it's not document being passed, it'll be another element that a listener can be attached to.

Reply:
I'm game.

Reply:
Is there a reason to have the last arguments grouped in an object? Should we turn it into one object? Or can we just add the one extra argument (document) at the end? Right now I have it as five individual arguments, but curious re: thoughts on this.

Reply:
Just realized the latest version of this does not work w/ the Menu component. Once Menu is in going to rebase and play with it there

Reply:
Also we need to rename this to `useEventListener` now as it can listen to non-document elements.

My thinking is the common way you'll use this is like: `useEventListener('resize', handler)`. If I want to use advanced features then I can opt-in via:

```
useEventListener('resize', handler, { element: window, enabled: false ))
```

I'd actually be fine having the argument signature be `(type, handler, options)` and move all advanced options to the hash at the end. Cuz type and handler are required and everything else is optional.